src/main/java/net/minecraftforge/network/VanillaConnectionNetworkFilter.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2020.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.network;
+
+import java.util.List;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToMessageEncoder;
+import net.minecraft.network.IPacket;
+import net.minecraft.network.NetworkManager;
+import net.minecraft.network.play.server.SEntityPropertiesPacket;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fml.network.NetworkHooks;
+import net.minecraftforge.registries.ForgeRegistries;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A filter for network packets, used to filter/modify parts of vanilla network messages that
+ * will cause errors or warnings on vanilla clients, for example entity attributes that are added by Forge or mods.
+ */
+public class VanillaConnectionNetworkFilter extends MessageToMessageEncoder<IPacket<?>>
+{
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    public static void injectIfNecessary(NetworkManager manager)
+    {
+        if (NetworkHooks.isVanillaConnection(manager))
+        {
+            manager.channel().pipeline().addBefore("packet_handler", "forge:vanilla_filter", new VanillaConnectionNetworkFilter());
+            LOGGER.debug("Injected into {}", manager);
+        }
+    }
+
+    @Override
+    protected void encode(ChannelHandlerContext ctx, IPacket<?> msg, List<Object> out)
+    {
+        IPacket<?> filtered;
+        if (msg instanceof SEntityPropertiesPacket)
```

Comment:
Can you make this a bit more pluggable? Put "handlers" as Functions into a map, and process the filter through that map. That will allow us to extend this behaviour to other packets we need to filter for -> vanillaclient traffic.